File: code/java/src/main/java/visitor/Museum.java
package visitor;

class Museum {
    public static void main(String[] args) {
        Exhibit painting = new Painting();
        Exhibit sculpture = new Sculpture();

        Visitor patron = new Patron();

        painting.accept(patron); // Patron interacts with a painting exhibit
        sculpture.accept(patron); // Patron interacts with a sculpture exhibit

    }
}

File: code/csharp/DesignPatterns/visitor/Museum.cs
namespace Visitor;

public class Museum
{
    public static void VisitorExample()
    {
        Exhibit painting = new Painting();
        Exhibit sculpture = new Sculpture();

        IVisitor patron = new Patron();

        painting.Accept(patron); // Patron interacts with a painting exhibit
        sculpture.Accept(patron); // Patron interacts with a sculpture exhibit
    }
}
